types/variant/write.C

```diff
@@ -0,0 +1,47 @@
+#include <ROOT/RNTupleModel.hxx>
+#include <ROOT/RNTupleWriteOptions.hxx>
+#include <ROOT/RNTupleWriter.hxx>
+
+using ROOT::Experimental::RNTupleModel;
+using ROOT::Experimental::RNTupleWriteOptions;
+using ROOT::Experimental::RNTupleWriter;
+
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include <variant>
+#include <vector>
+
+using Vector = std::vector<std::int32_t>;
+using Variant = std::variant<std::int32_t, std::string, Vector>;
+
+void write(std::string_view filename = "types.variant.root") {
+  auto model = RNTupleModel::Create();
+
+  auto value = model->MakeField<Variant>("f");
+
+  RNTupleWriteOptions options;
+  options.SetCompression(0);
+  auto writer =
+      RNTupleWriter::Recreate(std::move(model), "ntpl", filename, options);
+
+  // First entry: std::int32_t
+  *value = 1;
```

Comment:
Should we also test the case of `Fill` without a value set?

Reply:
@pcanal is this resolved?

Reply:
Yes. Thanks.